pkg/controller/appdefinition/deploy.go

```diff
@@ -553,6 +554,18 @@ func addPrometheusAnnotations(annotations map[string]string, container v1.Contai
 	}
 }
 
+func addCiliumAnnotations(annotations map[string]string, container v1.Container) {
+	var visibilityConfigs []string
+	for _, port := range container.Ports {
+		if port.Protocol == v1.ProtocolHTTP {
+			visibilityConfigs = append(visibilityConfigs, fmt.Sprintf("<Ingress/%d/TCP/HTTP>", port.TargetPort))
+		}
+	}
+	if len(visibilityConfigs) > 0 {
+		annotations[labels.CiliumProxyVisibility] = strings.Join(visibilityConfigs, ",")
```

Comment:
What is this annotion doing?

Reply:
These annotations tell Cilium to pay attention to this traffic and route it through the layer 7 proxy in order to generate metrics for it. It doesn't do this by default.